tests/codex/sales/testslotqueue.nim

```diff
@@ -515,9 +515,35 @@ suite "Slot queue":
       ]
     )
 
-  test "queue starts paused":
-    newSlotQueue(maxSize = 4, maxWorkers = 4)
-    check queue.paused
+  test "should process items in correct order according to the queue invariant when more than one item is added at a time":
+    newSlotQueue(maxSize = 4, maxWorkers = 2)
+    # sleeping after push allows the slotqueue loop to iterate,
+    # calling the callback for each pushed/updated item
+    var request = StorageRequest.example
+    let item0 = SlotQueueItem.init(request, 0)
+    request.ask.pricePerBytePerSecond += 1.u256
+    let item1 = SlotQueueItem.init(request, 1)
+    request.ask.pricePerBytePerSecond += 1.u256
+    let item2 = SlotQueueItem.init(request, 2)
+    request.ask.pricePerBytePerSecond += 1.u256
+    let item3 = SlotQueueItem.init(request, 3)
+
+    check queue.push(item0).isOk
+    check queue.push(item1).isOk
+    check queue.push(item2).isOk
+    check queue.push(item3).isOk
+
+    await sleepAsync(1.millis)
+
+    check eventually (
+      onProcessSlotCalledWith ==
+      @[
+        (item3.requestId, item3.slotIndex),
+        (item2.requestId, item2.slotIndex),
+        (item1.requestId, item1.slotIndex),
+        (item0.requestId, item0.slotIndex),
+      ]
+    )
```

Comment:
IIRC, i had issues with this passing consistently on slow CI systems due to popping of SlotQueueItems before the new ones are added. Hopefully this doesn't happen 👍

Reply:
Oh, I see - so far it seems to be working ok. Thanks!